gneiss/_model.py

```diff
@@ -0,0 +1,122 @@
+import pickle
+from skbio import TreeNode
+import abc
+from skbio.stats.composition import ilr_inv, closure
+import numpy as np
+import pandas as pd
+
+
+class Model(metaclass=abc.ABCMeta):
+
+    def __init__(self, submodels, basis, tree, balances):
+        """
+        Abstract container for balance models.
+
+        Parameters
+        ----------
+        submodels : list of statsmodels objects
+            List of statsmodels result objects.
+        basis : pd.DataFrame
+            Orthonormal basis in the Aitchison simplex.
+            Row names correspond to the leafs of the tree
+            and the column names correspond to the internal nodes
+            in the tree. If this is not specified, then `project` cannot
+            be enabled in `coefficients` or `predict`.
+        tree : skbio.TreeNode
+            Bifurcating tree that defines `basis`.
+        balances : pd.DataFrame
+            A table of balances where samples are rows and
+            balances are columns.  These balances were calculated
+            using `tree`.
+        """
+        self.submodels = submodels
+        self.basis = basis
+
+        # Make all nodes in the tree queriable
+        self._nodes = {n.name: n for n in tree.levelorder()}
+
+        self.tree = tree
+        self.balances = balances
+        self.results = []
+
+    def fit(self, **kwargs):
+        """ Fit the model """
+        for s in self.submodels:
+            # assumes that the underlying submodels have implemented `fit`.
+            m = s.fit(**kwargs)
+            self.results.append(m)
+
+    @abc.abstractmethod
+    def summary(self):
+        """ Print summary results """
+        pass
+
+    def split_balance(self, balance_name):
+        """ Splits a balance into its log ratio components.
+
+        Parameters
+        ----------
+        node : str
```

Comment:
I'd prefer to add the code for throwing an Error in this PR rather than waiting for future changes. Since it might slip through your to-do list.